app/angular/src/server/index.ts

```diff
@@ -1,4 +1,6 @@
 import { buildDev } from '@storybook/core/server';
+import { runNgcc } from './ngcc-execution';
 import options from './options';
 
+runNgcc();
```

Comment:
This is a synchronous operation, correct?

Reply:
You should be aware that in sync mode, ngcc will crash out if there is another ngcc process running at the same time - it is not able to wait like it can in async mode. This is probably not a problem but worth being aware of.

Reply:
Thanks for the info! I can imagine that someone starts the app + storybook in parallel right after installing all dependencies
Using async should cause no further issues, right? We just need to `await` ngcc 

If it is that simple to prevent a crash then let's do this 🙂

Reply:
If you can call `runNgcc()` asynchronously then it is probably safest to do so.

Reply:
@petebacondarwin I'm running into a strange behavior using `async: true`

When I run my dev environment with this
```
// important part is `-- --no-manager-cache --quiet`
"start": "concurrently \"npm run storybook -- --no-manager-cache --quiet\" \"npm run build -- --watch\"",
```

I get this error

```
[0] Options:
[0]       --version                          Show version number           [boolean]
[0]   -s, --source                           A path (relative to the working
[0]                                          directory) of the `node_modules` folder
[0]                                          to process.
[0]                                             [string] [default: "./node_modules"]
[0]   -p, --properties                       An array of names of properties in
[0]                                          package.json to compile (e.g. `module`
[0]                                          or `main`)
[0]                                          Each of these properties should hold
// etc. etc.

[0] Unknown arguments: manager-cache, managerCache, quiet

[0] Warning: Worker #1 exited unexpectedly (code: 1 | signal: null).
[0]   Current task: {entryPoint: @angular/platform-browser-dynamic, formatProperty: fesm2015, processDts: Yes}
[0]   Current phase: compiling
```

It passes the arguments for `run storybook` to the ngcc `process` execution 

```
[0] Unknown arguments: manager-cache, managerCache, quiet
```

Is there anything I can to do avoid this? 

Running it with 

```
"start": "concurrently \"npm run storybook\" \"npm run build -- --watch\"",
```

does work as expected (and async: true)

Reply:
@shilman @ndelangen or is this something that happens within the storybook startup?

Reply:
I don't know. I don't think Storybook is doing anything special with respect to passing extra arguments through.

Reply:
That error indicates that you are somehow triggering ngcc via its command line executable.
I don't suppose you have a `prestorybook` or `prebuild` script or something that is still running `ngcc` from the command line?

Reply:
No, only from the code. I extracted the code from this PR in a smaller external package which makes it easier to see what happens: https://github.com/storybookjs/addon-angular-ivy/blob/main/src/preset/index.ts

That's all

There are 2 ways to avoid the error
- Use `async: false` 
- Use `async: true` and don't add arguments to the `"start"` script 

No other prebuild steps involved: https://github.com/storybookjs/addon-angular-ivy/blob/main/package.json#L22

Reply:
Oh I just realised... in async mode we kick off child processes, which will receive the arguments passed to the original parent process. Perhaps this is how they are getting passed to ngcc. I'll investigate

Reply:
Perhaps this is where it is coming from?

https://github.com/angular/angular/blob/6de5a12a9dc70dbbd226693ca7e378f813b213c8/packages/compiler-cli/ngcc/src/execution/cluster/worker.ts#L36

Reply:
A workaround would be to clear the command line args before triggering ngcc:

```ts
process.argv.length = 0;
```